platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,259 @@
+
+    protected static final String ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG = "Error retrieving attributes from external attribute store [%s] for DN [%s]. ";
+
+    private String attributeMapLocation;
+
+    private List<String> supportedClaims;
+
+    private Map<String, String> attributeMap;
+
+    protected SimpleSign simpleSign;
+
+    protected String externalAttributeStoreUrl;
+
+    protected String issuer;
+
+    protected String destination;
+
+    public AttributeQueryClaimsHandler() {
+        LOGGER.debug("Starting AttributeQueryClaimsHandler");
+    }
+
+    /**
+     * Gets the supported claim types.
+     *
+     * @return List of supported claims.
+     */
+    @Override
+    public List<URI> getSupportedClaimTypes() {
+        LOGGER.debug("Getting supported claim types.");
+        List<URI> supportedClaimTypes = new ArrayList<>();
+        try {
+            for (String claim : supportedClaims) {
+                supportedClaimTypes.add(new URI(claim));
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.warn("Not a valid URI for claim type {}.", e);
+        }
+        return supportedClaimTypes;
+    }
+
+    /**
+     * Retrieves claims from the external attribute store.
+     *
+     * @param claims The collection of claims.
+     * @return The collection of claims or null if there are no security claims.
+     * @throws URISyntaxException
+     */
+    @Override
+    public ProcessedClaimCollection retrieveClaimValues(ClaimCollection claims,
+            ClaimsParameters parameters) {
+        Principal principal = parameters.getPrincipal();
+
+        String username = getUsername(principal);
+
+        ProcessedClaimCollection claimCollection = new ProcessedClaimCollection();
+        try {
+            if (!StringUtils.isEmpty(username)) {
+                ProcessedClaimCollection securityClaimCollection = this.getAttributes(username);
+                // If security claim collection came back empty, return an empty claim collection.
+                if (!CollectionUtils.isEmpty(securityClaimCollection)) {
+                    claimCollection.addAll(securityClaimCollection);
+                }
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error(ERROR_RETRIEVING_ATTRIBUTES, externalAttributeStoreUrl, username, e);
+            SecurityLogger.logError(
+                    String.format(ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG, externalAttributeStoreUrl,
+                            username), e);
+        }
+
+        return claimCollection;
+    }
+
+    /**
+     * Parses the CN from the user's DN.
+     *
+     * @param principal of the user.
+     * @return CN of the user.
+     */
+    protected String getUsername(Principal principal) {
+        return Pattern.compile(",").splitAsStream(principal.getName())
+                .filter(s -> s.startsWith("CN=".toUpperCase())).findFirst().map(s -> s.substring(3))
+                .orElse(null);
+
```

Comment:
You can use bouncy castle for this.

    X500Name blah = new X500Name(principal.getName())
    String cn = blah.getRDNs(BCStyle.CN)[0];